core/domain/improvements_domain.py

```diff
@@ -0,0 +1,243 @@
+                entity_id: str. The ID of the entity a task entry refers to.
+                entity_version: int. The version of the entity a task entry
+                    refers to.
+                task_type: str. The type of task a task entry tracks.
+                target_type: str. The type of sub-entity a task entry refers to.
+                target_id: str. The ID of the sub-entity a task entry refers to.
+                issue_description: str. The sentence generated by Oppia to
+                    describe why the task was created.
+                status: str. Tracks the state/progress of a task entry.
+                resolver_username: str. Username of the user who resolved the
+                    task, if any.
+                resolved_on_msecs: float. Time this task was closed on in
+                    milliseconds since epoch.
+        """
+        return {
+            'entity_type': self.entity_type,
+            'entity_id': self.entity_id,
+            'entity_version': self.entity_version,
+            'task_type': self.task_type,
+            'target_type': self.target_type,
+            'target_id': self.target_id,
+            'issue_description': self.issue_description,
+            'status': self.status,
+            'resolver_username': (
+                self.resolver_id and
+                user_services.get_username(self.resolver_id)),
+            'resolved_on_msecs': (
+                self.resolved_on and
+                utils.get_time_in_millisecs(self.resolved_on)),
+        }
+
+    @classmethod
+    def from_dict(cls, task_dict):
+        """Constructs a new domain instance from given dict of values. Ignores
+        fields related to status (resolved_on_msecs, resolver_username), due to
+        their vulnerability to becoming out of sync.
+
+        Args:
+            task_dict: dict. Contains the following keys:
+                entity_type: str. The type of entity a task entry refers to.
+                entity_id: str. The ID of the entity a task entry refers to.
+                entity_version: int. The version of the entity a task entry
+                    refers to.
+                task_type: str. The type of task a task entry tracks.
+                target_type: str. The type of sub-entity a task entry refers to.
+                target_id: str. The ID of the sub-entity a task entry refers to.
+                issue_description: str. The sentence generated by Oppia to
+                    describe why the task was created.
+                status: str. Tracks the state/progress of a task entry.
+
+        Returns:
+            TaskEntry.
+        """
+        return cls(
+            task_dict['entity_type'], task_dict['entity_id'],
+            task_dict['entity_version'], task_dict['task_type'],
+            task_dict['target_type'], task_dict['target_id'],
+            task_dict['issue_description'], task_dict['status'], None, None)
+
+    def to_model(self):
```

Comment:
Please follow existing code conventions. Domain files should be standalone and should not be importing the storage layer.